### PR TITLE
Fix asset release

### PR DESCRIPTION
**Pull request by novaturient95**

### CODE REVIEW

File: scripts/release/asset_publish.py [hunk 1]
```diff
@@ -166,17 +167,28 @@ def validate_update_command_component(
...
     env = component.environment
     match = None
-    for pattern in [REGISTRY_ENV_PATTERN, WORKSPACE_ASSET_PATTERN]:
-        if (match := pattern.match(env)) is not None:
-            break
+    if (match := REGISTRY_ENV_PATTERN.match(env)) is not None:
+        env_registry_name, env_name, env_version, env_label = (
+            match.group(1), match.group(2), match.group(3), match.group(4))
+    elif (match := WORKSPACE_ASSET_PATTERN.match(env)) is not None:
+        env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
```
Comment (elesel) [inline scripts/release/asset_publish.py:174]:
```suggestion
    elif (match := WORKSPACE_ASSET_PATTERN.match(env)) is not None:
        env_registry_name = None
        env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
```

Comment (novaturient95) [reply scripts/release/asset_publish.py:175]: Done. Thank you

File: scripts/release/asset_publish.py [hunk 2]
```diff
@@ -166,17 +167,29 @@ def validate_update_command_component(
...
         return False
 
-    # both ws and registry env follows the same grouping
-    env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
-    logger.print(f"Env name: {env_name}, version: {env_version}, label: {env_label}")
+    logger.print(
+        f"Env name: {env_name}, version: {env_version}, label: {env_label}, env_registry_name: {env_registry_name}"
+    )
+
+    if env_registry_name and env_registry_name != PROD_SYSTEM_REGISTRY and env_registry_name != registry_name:
```
Comment (elesel) [inline scripts/release/asset_publish.py:185]:
```suggestion
    if env_registry_name and env_registry_name not in [PROD_SYSTEM_REGISTRY, registry_name]:
```

File: scripts/release/asset_publish.py [hunk 3]
```diff
@@ -166,17 +167,29 @@ def validate_update_command_component(
...
+        logger.log_warning(
+            "Unexpected !!! Registry name for component's env URI must be either "
+            + f"'{registry_name}' or '{PROD_SYSTEM_REGISTRY}'. Got '{env_registry_name}'"
+        )
+        return False
+
+    registry_name = registry_name if not env_registry_name else env_registry_name
```
Comment (elesel) [inline scripts/release/asset_publish.py:192]:
```suggestion
    registry_name = env_registry_name or registry_name
```

File: scripts/release/asset_publish.py [hunk 4]
```diff
@@ -166,17 +167,29 @@ def validate_update_command_component(
...
+    env_registry_name = None
+    if (match := REGISTRY_ENV_PATTERN.match(env)) is not None:
+        env_registry_name, env_name, env_version, env_label = (
+            match.group(1), match.group(2), match.group(3), match.group(4))
+    elif (match := WORKSPACE_ASSET_PATTERN.match(env)) is not None:
+        env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
 
     if not match:
         logger.print(f"Env ID doesn't match workspace or registry pattern in {asset.spec_with_path}")
         return False
```
Comment (elesel) [inline scripts/release/asset_publish.py:179]:
```suggestion
        env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
    else:
        logger.print(f"Env ID doesn't match workspace or registry pattern in {asset.spec_with_path}")
        return False
```

Comment (novaturient95) [reply scripts/release/asset_publish.py:179]: address comments

**Review by elesel — APPROVED** [review]

**Pull request merged**